pkg/southbound/synchronizer/synchronizer.go

```diff
@@ -264,36 +261,133 @@ func (sync Synchronizer) syncOperationalState(errChan chan<- events.DeviceRespon
 	//  wildcarded paths - assuming that stratum does not support wildcarded GET
 	//  and SUBSCRIBE)
+	log.Infof("%d ReadOnly paths for %s", len(sync.modelReadOnlyPaths), sync.key.DeviceID)
+	if len(sync.modelReadOnlyPaths) == 0 {
+		noPathErr := fmt.Errorf("target %#v has no paths to subscribe to", sync.ID)
+		errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorSubscribe,
+			sync.key.DeviceID, noPathErr)
+		log.Warning(noPathErr)
+		return
+	}
+
+	wildExpandedPaths := make([]string, 0)
+	if len(notifications) == 0 {
+		log.Infof("No notifications received - trying Get on the readonly paths instead %s", sync.key.DeviceID)
+		getPaths := make([]*gnmi.Path, 0)
+		for _, path := range modelregistry.Paths(sync.modelReadOnlyPaths) {
+			gnmiPath, err := utils.ParseGNMIElements(utils.SplitPath(path))
+			if err != nil {
+				log.Warning("Error converting RO path to gNMI")
+				errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorTranslation,
+					sync.key.DeviceID, err)
+				return
+			}
+			getPaths = append(getPaths, gnmiPath)
+		}
+
+		requestRoPaths := &gnmi.GetRequest{
+			Encoding: sync.encoding,
+			Path:     getPaths,
+		}
+
+		responseRoPaths, errRoPaths := target.Get(target.Ctx, requestRoPaths)
+		if errRoPaths != nil {
+			log.Warning("Error on request for read-only paths", sync.key, errRoPaths)
+			errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorGetWithRoPaths,
+				sync.key.DeviceID, errRoPaths)
+			return
+		}
+
+		notifications = append(notifications, responseRoPaths.Notification...)
+
+		////////////////////////////////////////////////////////////////////////
+		// Special case for Stratum Aug'19 - the notification will only contain
+		// the ifindex and name of the interface - to get the other state attributes
+		// it has to be called again with the wildcard values expanded out. This
+		// is because Stratum does not support wildcards properly.
+		// Fortunately these can be also used for subscribe
+		////////////////////////////////////////////////////////////////////////
+		if sync.Type == "Stratum" {
+			ewGetPaths := make([]*gnmi.Path, 0)
+			for _, n := range responseRoPaths.Notification {
+				for _, u := range n.Update {
+					updatepath := utils.StrPath(u.Path)
+					if strings.HasPrefix(updatepath, "/interfaces/interface[") {
+						newPathStr := updatepath[:strings.LastIndex(updatepath, "/ifindex")]
+						wildExpandedPaths = append(wildExpandedPaths, newPathStr)
+						newGnmiPath, errNewPath := utils.ParseGNMIElements(utils.SplitPath(newPathStr))
+						if errNewPath != nil {
+							log.Warning("Error on request for read-only paths", sync.key, errNewPath)
+							errChan <- events.CreateErrorEventNoChangeID(events.EventTypeErrorGetWithRoPaths,
+								sync.key.DeviceID, errNewPath)
+							return
+						}
+						log.Info("Making special expanded wildcard call to ", newPathStr, " for ", sync.key)
+						ewGetPaths = append(ewGetPaths, newGnmiPath)
```

Comment:
it seems that you are not removing the wilcarded path from the the ewGetPaths. That could issue problems.

Reply:
I don't follow. BTW I was only adding to wildcarded paths so I have them afterwards for "subscribe"

Reply:
yes, I was wondering if you also had other * path other than interfaces. I'm ok to handle that use case later, please add a TODO and/or open an issue.